packages/headless/src/controllers/facets/facet/headless-facet.ts

```diff
@@ -296,12 +316,20 @@ export function buildFacet(engine: Engine<object>, props: FacetProps): Facet {
     toggleSelect: (selection: FacetValue) =>
       dispatch(executeToggleFacetSelect({facetId: options.facetId, selection})),
 
+    toggleSingleSelect: (selection: FacetValue) => {
+      if (selection.state !== 'idle') {
```

Comment:
Here's how I mitigated the "toggle problem" discussed in this thread https://github.com/coveo/ui-kit/pull/874#discussion_r648559754

We would like to click on a single selected value see it deselected, for this to happen, we should not call toggle.
The analytic here could be a problem, I could instead dispatch a search with the logFacetDeselect event, what do you think?

Reply:
Small downside: A customer not using the controller would still have to figure this out.

Reply:
And the rework for the original PR was just about us having less code ? 🤔 

Yeah it does have the drawback of making it so users * have * to use the controller, which is so so, but not the end of the world.

To figure it out, I guess just looking at the code and mimicking it will be an alternative 😂

Reply:
Yes it was a requested change to remove the actions and have less source code, which with hindsight I agree with

Reply:
A simplification would be to not handle the deselect case inside of the method, reducing the scope from `toggleSingleSelect` to just `singleSelect`, like the facet search.

If someone wants to implement a radio-list UX, they would use the new `singleSelect` method.

```
function onClick(...) {
    singleSelect(facetValue);
}
```

If someone wants to implement a single-select checkbox UX (as in our case), then the UI can leverage the controller like this:

```
function onClick(...) {
    facetValue.state=== 'idle' ? singleSelect(facetValue) : toggleSelect(facetValue);
}
```

When I look at the code, I see quite some complexity being added to manage the deselect case. Maybe the value provided justifies the cost, but I think it's debatable. With just `singleSelect`, the controller is still providing flexibility, while keeping the code base simple for future maintainers.

Just giving a different perspective on the problem. I am fine with either decision,

Reply:
Good point, kept the behaviour, but simplified the implementation.